### PR TITLE
Add performance boost

### PR DESCRIPTION
**Pull request by Poltergeist**
This adds a performance hotfix which has been created by @orlandohohmeier.

 
image

image

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @kennyt

File: src/js/components/modals/ServiceFormModal.js [hunk 1]
```diff
@@ -135,6 +136,13 @@ class ServiceFormModal extends mixin(StoreMixin) {
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    let {state, props} = this;
+    return props.open !== nextProps.open ||
+      state.jsonMode !== nextState.jsonMode ||
+      state.errorMessages !== nextState.errorMessage;
+  }
```
Comment (pierlo-upitup) [inline src/js/components/modals/ServiceFormModal.js]: could there be a case where the user didn't fix the error message and we need to re-render it?

Comment (orlandohohmeier) [inline src/js/components/modals/ServiceFormModal.js]: We're re-rendering if the `errorMessage` is updated (e.g. cleared) or (!) the `jsonMode` is toggled or the modal open property is toggled. Every other case should be handled by the child components.

Comment (Poltergeist) [inline src/js/components/modals/ServiceFormModal.js]: the errorMessage gets set to null on change (reads: focus/blur/change) thus we have to re-render.

File: src/js/components/modals/ServiceFormModal.js [hunk 2]
```diff
@@ -135,6 +136,13 @@ class ServiceFormModal extends mixin(StoreMixin) {
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    let {state, props} = this;
+    return props.open !== nextProps.open ||
+      state.jsonMode !== nextState.jsonMode ||
+      state.errorMessages !== nextState.errorMessage;
```
Comment (kennyt) [inline src/js/components/modals/ServiceFormModal.js]: here it's comparing `errorMessages` with `errorMessage`. the second one is not plural. is that what we want?

**Comment by kennyt** [comment]
i'd merge after my comment is addressed

**Comment by rcorral** [comment]
retest this please

File: src/js/components/modals/ServiceFormModal.js (hunk 2 above)
Comment (Poltergeist) [inline src/js/components/modals/ServiceFormModal.js]: Good catch

**Comment by Poltergeist** [comment]
image

**Comment by Poltergeist** [comment]
**mesosphere-ci** retest this please

**Comment by Poltergeist** [comment]
**mesosphere-ci** retest this please

File: src/js/components/modals/ServiceFormModal.js [hunk 3]
```diff
@@ -166,7 +174,7 @@ class ServiceFormModal extends mixin(StoreMixin) {
     } catch (e) {
 
     }
-    this.setState({jsonDefinition, service})
+    this.setState({jsonDefinition, service});
```
Comment (aldipower) [inline src/js/components/modals/ServiceFormModal.js]: Strike!

**Pull request merged**